### ISSUE TITLE
[49.7] Campaign Fails to Load After Aerospace Unit Is Purchased

### ISSUE THREAD

**Issue by PhoenixHeart512** [MegaMek/mekhq#3199]
### Environment
MekHQ 0.49.7
Java Vendor: Eclipse Adoptium
Java Version: 11.0.14.1
Win10 21h2

### Description
After purchasing any Aerospace unit via "Purchase Units" (tried 3 different ones, the Stuka STU-D6, the Huscarl HSCL-O, and Corax CRX-O) and saving my game, if I try to load that save file it fails to load with the error shown in the attached screenshot.

This only occurs if the unit is purchased through the "purchase unit" menu. I was able to buy an ASF (Gotha GTHA-500) through the monthly "unit market" and my game successfully saved/loaded. I then bought the exact same ASF via "purchase unit" and upon saving and loading, the error occurred and I could no longer load my game.

I tried enabling the "use aerospace" option in the AtB Campaign Options tab, but the issue still occurred.

Steps to reproduce:
1. Load save file Tempest Ghosts.cpnx.gz
2. Click Marketplace -> Purchase Unit -> Aerospace -> pick any ASF (Huscarl-O used for testing, stock build, not custom)
3. Save Game (Preferably as new file to make it easier to repeat the process later)
4. Load the campaign save created in step 3, error will occur during loading screen

Tempest Ghosts TEMP.cpnx.gz save file is included as an already-broken sample, but not necessary to replicate the issue.

Logs and files included in the attached zip, folder structure mirrors a normal MekHQ installation, so if you extract the contents into a MekHQ directory, it should install and overwrite everything into the correct locations (settings and preferences file included).

### Files
[Campaign Load Fails After Aerospace Purchase.zip](https://github.com/MegaMek/mekhq/files/8393498/Campaign.Load.Fails.After.Aerospace.Purchase.zip)

**Comment by PhoenixHeart512** [comment]
Actually I believe this is something on my end, maybe a corrupt installation, as I am no longer able to buy ANY units through the "purchase unit" dropdown, even units that I have purchased before in this campaign. This is also happening even if I create a new campaign.

I believe this bug report can be closed, unless someone else is able to repro this behavior as well.

**Comment by PhoenixHeart512** [comment]
And I just found https://github.com/MegaMek/mekhq/issues/3139 even though I thought I had searched reported issues already, apparently not well enough. I also see the workaround, and that actually perfectly explains why it's only affecting units purchased through the "purchase units" window, because those units are "ordered", unlike units bought off market.

This ticket can definitely be closed now lol. Do I do that myself, or can only devs close issues?
